### PR TITLE
Switch to using flit for packaging

### PR DESCRIPTION
**Pull request by zerolab**
Builds on #116

### CODE REVIEW

File: .github/workflows/publish.yml
Comment (Stormheg) [inline .github/workflows/publish.yml:38]: **Question:** perhaps we should consider setting up trusted publishing? Like done for wagtail-bakery: https://github.com/wagtail-nest/wagtail-bakery/commit/0bfff2a7446e6119156279d22cfac2137960145d

File: README.md
Comment (Stormheg) [inline README.md:85]: Thanks for converting this to markdown!

File: README.md
Comment (Stormheg) [inline README.md:1]: Markdown rendering is broken

```suggestion
# [Willow image library](https://pypi.org/project/wagtailmedia/)
```

File: pyproject.toml
Comment (Stormheg) [inline pyproject.toml:68]: Happy to see this converted to pyproject.toml!

File: README.md
Comment (Stormheg) [inline README.md:5]: Yay to badges!

File: README.md
Comment (zerolab) [reply README.md]: sorted. I copied this and the badges from wagtailmedia and the above URL was wrong 🙈 😂

File: .github/workflows/publish.yml
Comment (zerolab) [reply .github/workflows/publish.yml]: Done in 648492c

File: .github/workflows/test.yml
Comment (Stormheg) [inline .github/workflows/test.yml:26]: @zerolab would you document this change in the changelog, please?

**Review by Stormheg — APPROVED** [review]
All looks good to me. Haven't tested the build process but trust it works like it should.

Could you please document dropping Python 3.7 support in the changelog before merging?

File: .github/workflows/test.yml
Comment (zerolab) [reply .github/workflows/test.yml:26]: it is already in 1.5 - https://github.com/wagtail/Willow/blob/main/docs/changelog.rst?plain=1#L7

Comment (Stormheg) [reply .github/workflows/test.yml:26]: @zerolab that doesn't state Python 3.7 support was dropped. It states support for all versions _before_ 3.7 is dropped.

Comment (Stormheg) [reply .github/workflows/test.yml:26]: The wording in that changelog entry is a bit ambitious, easily misinterpreted

Comment (zerolab) [reply .github/workflows/test.yml:26]: 🤦🏼 yup. Can you tell I am too eager?

Comment (Stormheg) [reply .github/workflows/test.yml:26]: That's why having a second pair of eyes is helpful 🙂

File: pyproject.toml
Comment (Stormheg) [inline pyproject.toml:4]: Just wondering: should we add a [`maintainers`](https://peps.python.org/pep-0621/#authors-maintainers) key and list the Wagtail Core team as maintainers? Since Karl has left the Wagtail space.

File: .github/workflows/test.yml
Comment (zerolab) [reply .github/workflows/test.yml:26]: 5de38a6

File: pyproject.toml
Comment (zerolab) [reply pyproject.toml:4]: It is a good suggestion, but let's tweak this in a later PR as I don't have a good list :D

Comment (Stormheg) [reply pyproject.toml:4]: I did not mean to list all members of the team since people come and go all the time. I meant a more generic `Wagtail Core team <hello@wagtail.org>` entry.

Comment (zerolab) [reply pyproject.toml:4]: ah! will address separately

Comment (zerolab) [reply pyproject.toml:4]: 165078e

**Pull request merged**
